dbt/adapters/impala/connections.py

```diff
@@ -159,27 +160,58 @@ def open(cls, connection):
                     port=credentials.port,
                     retries=credentials.retries,
                 )
+            connection_end_time = time.time()
 
-            connection.state = "open"
+            connection.state = ConnectionState.OPEN
             connection.handle = handle
         except Exception as ex:
             LOGGER.debug("Connection error {}".format(ex))
-            connection.state = "fail"
+            connection.state = ConnectionState.FAIL
             connection.handle = None
+            connection_end_time = time.time()
             pass
 
         # track usage
         payload = {
             "id": "dbt_impala_open",
-            "unique_hash": hashlib.md5(credentials.host.encode()).hexdigest(),
+            "unique_hash": hashlib.md5((str(credentials.host) + str(credentials.username)).encode()).hexdigest(),
             "auth": auth_type,
+            "connection_name": connection.name,
             "connection_state": connection.state,
+            "elapsed_time": "{:.2f}".format(connection_end_time - connection_start_time),
```

Comment:
If we are opening a connection here, elapsed_time doesn't make any sense.

Reply:
elapsed_time for opening a connection, that is what is the interpretation.

Reply:
nvm.. I understood. This was confusing for me a bit because of the variable name. Nitpick please rename it to connection_duration or duration_of_connection maybe